### ISSUE TITLE
Telemetry ignores all events if a trip session is started before a route is set

### ISSUE THREAD

**Issue by LukasPaczos** [mapbox/mapbox-navigation-android#3135]
The [`MapboxNavigationTelemetry.kt`](https://github.com/mapbox/mapbox-navigation-android/blob/master/libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt) is set up in a way, that forces the route to be present _before_ we start the trip session, otherwise, all of the telemetry events are ignored.
The SDK makes this check:
https://github.com/mapbox/mapbox-navigation-android/blob/b469ee786e60bef101c4250774f85037abc9d7f4/libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt#L172-L177
https://github.com/mapbox/mapbox-navigation-android/blob/b469ee786e60bef101c4250774f85037abc9d7f4/libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt#L237-L239
but the route is only assigned when the trip session is started:
https://github.com/mapbox/mapbox-navigation-android/blob/b469ee786e60bef101c4250774f85037abc9d7f4/libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt#L493-L502

In short, this means that if the trip session is started before the route is set, we will not send any telemetry events.

@mapbox/navigation-android

**Comment by Guardiola31337** [comment]
This is definitely related to https://github.com/mapbox/mapbox-navigation-android/issues/2651 and handling the transition between _Free Drive_ and _Active Guidance_ (currently there's basically no transition 😬).

cc @LukasPaczos
